src/SourceTerm/typedefs.hpp

```diff
@@ -46,9 +47,73 @@
 #include <vector>
 #include <Initializer/typedefs.hpp>
 #include <generated_code/tensor.h>
+#include <cstdint>
+
+#ifdef ACL_DEVICE
+#include <sycl/sycl.hpp>
+#endif
 
 namespace seissol {
-  namespace sourceterm {    
+  namespace sourceterm {
+#ifdef ACL_DEVICE
+    template <typename T>
+    using AllocatorT = sycl::usm_allocator<T, sycl::usm::alloc::shared>;
+    class AllocatorFactory {
+    public:
+      AllocatorFactory(sycl::queue q) : q_(std::move(q)) {}
+      template <typename T> AllocatorT<T> make() const { return AllocatorT<T>(q_); }
+    private:
+      sycl::queue q_;
+    };
+#else
```

Comment:
Hi, this is defined in the header file. It means that all files which explicitly or implicitly include it must to be compiler with a `sycl` compiler. It is totally ok for DPC++ compiler but I think it is not going to work for OpenSYCL (we are using separate compilation for the host and device codes).

For example, I haven't managed to install DPC++ on LUMI (Cray Machine) and thus I have only one alternative - i.e., OpenSYCL.

What do you think if we use our `device` wrapper in this part of the code?  

```
device.api->allocUnifiedMem(size * sizeof(T))
```

I think we just need to create `AllocatorT` for it.

Reply:
I don't mind using the device wrapper but usm_allocator is part of the SYCL spec and also included in opensycl: https://github.com/OpenSYCL/OpenSYCL/blob/12fdcaedfa990ab58ddf8bce304fa8cf917e6182/include/hipSYCL/sycl/usm.hpp#L270

Let me try whether I can compile with opensycl as well